### PR TITLE
Change LogAcceptCategory to use uint32_t rather than sets of strings.

### PR DESCRIPTION
**Pull request by dagurval**
https://github.com/bitcoin/bitcoin/pull/9424

I changed namespace BCLog to Log

### CODE REVIEW

**Comment by dgenr8** [comment]
Running bitcoin-qt in regtest with -debug=all, I'm getting messages like this in debug.log:
<pre>2018-03-29 05:17:06   1: For conf success > 0.85 need FeeRate >:           -1 from buckets  2.1e+15 -  2.1e+15  Cur Bucket stats   -nan%       0.0/(0.0+0 mempool)</pre>
It seems to be printing the wrong debug category and also not translating the number to a name. I haven't been able to figure it out yet.

File: src/qt/bitcoin.cpp [hunk 1]
```diff
@@ -149,15 +149,21 @@ static void initTranslations(QTranslator &qtTranslatorBase, QTranslator &qtTrans
 #if QT_VERSION < 0x050000
 void DebugMessageHandler(QtMsgType type, const char *msg)
 {
-    const char *category = (type == QtDebugMsg) ? "qt" : NULL;
-    LogPrint(category, "GUI: %s\n", msg);
+    if (type == QtDebugMsg) {
+        Log(Log::QT, "GUI: %s\n", msg);
```
[the remark this answers is not in the document]
Comment (dagurval) [reply src/qt/bitcoin.cpp]: Fixed. I guess I'll wait another year before posting a remove-qt4 PR ;-)

**Comment by dagurval** [comment]
From quick review of that code, that log entry looks correct under error conditions.

It's not supposed to log the debug category.

**Comment by dgenr8** [comment]
Oh, right. Duh.

**Pull request merged**